src/ucs/debug/memtrack.c

```diff
@@ -201,11 +200,19 @@ void *ucs_realloc(void *ptr, size_t size, const char *name)
     return ptr;
 }
 
-void *ucs_memalign(size_t boundary, size_t size, const char *name)
+int ucs_posix_memalign(void **ptr, size_t boundary, size_t size, const char *name)
 {
-    void *ptr = memalign(boundary, size);
-    ucs_memtrack_allocated(ptr, size, name);
-    return ptr;
+    int ret;
+
+#if HAVE_POSIX_MEMALIGN
+    ret = posix_memalign(ptr, ucs_max(boundary, sizeof(void *)), size);
```

Comment:
think of it - this is not the right place. the **users** of posix_memalign should make sure to pass the correct value.
otherwise, it would not help when memtrack is disabled in build config

Reply:
Why would users of the ucs_posix_memalign() need to care about size of the pointer ?  IMO it is the right place to satisfy the interface requirements.  For non-debug build, the same ucs_max(sizeof(void *), ...) op is performed.

Reply:
ucs_posix_memalign() should have same API requirements as posix_memalign()
because in non-debug build, ucs_posix_memalign is defined to be posix_memalign()

Reply:
For non-debug builds, ucs_posix_memalign() is not defined to posix_memalign(). Please look at its actual definition.

Reply:
i see it, so this is another thing to fix: ucs_posix_memalign should be defined as posix_memalign.